### ISSUE TITLE
Support seeds for trials

### ISSUE THREAD

**Issue by bouthilx** [Epistimio/orion#71]
From comment [here](https://github.com/mila-udem/orion/pull/69#discussion_r186903835)

> Two trials with identical parameters could have different results. We can't assume the user's script is deterministic. We could, however, add a seed to the attributes of the Trial and pass it through an environment variable to the user's script. The user could seed the script using it, making it deterministic for a given pair of (params, seed).

Trials should be assigned random seeds, which are passed to user script using exposed environment variables. There should also be an option to specify if an experiment is noisy or deterministic. In the deterministic case, there would be no seed, or a single seed for the entire experiment.

**Comment by tsirif** [comment]
Copy pasting my answer from #69 

> `hash_name` is supposed to be generated and given as an input to user's script so that it is able to disambiguate local directories that it may use (i.e. for snapshots or logs) among different hyperparameter trial sets. It is supposed to be generated by trials whose results are not yet known, as is the use case above. So this hash_name corresponds only to `trial.params` only. To make it able to run multiple experiments for the same set of hyperparameters:
> 1. yes we should provide the seed to the user's script. I predict a `~trial.seed` kind of command in the likes of what I've built here. So, `seed` is going to be an attribute of a `Trial` as well.
> 2. `hash_name` will be generated as a function of `(trial.params, trial.seed)` to ensure uniqueness of results and reproducibility.

**Comment by bouthilx** [comment]
Yes this is what I meant. Once the hash_name is generated in function of (trial.params, trial.seed) it means trials with identical parameters but different seed will have different local directories.

**Comment by bouthilx** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Support seeds for trials](https://trello.com/c/ziP8dxOj/239-support-seeds-for-trials)

**Comment by bouthilx** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Support seeds for trials](https://trello.com/c/ziP8dxOj/239-support-seeds-for-trials)
